aptos-move/block-executor/src/txn_last_input_output.rs

```diff
@@ -242,9 +269,27 @@ impl<T: Transaction, O: TransactionOutput<Txn = T>, E: Debug + Send + Clone>
         )
     }
 
+    pub(crate) fn record_finalized_groups(
+        &self,
+        txn_idx: TxnIndex,
+        finalized_groups: Vec<(T::Key, T::Value, Vec<(T::Tag, Arc<T::Value>)>)>,
+    ) {
+        match &self.outputs[txn_idx as usize]
+            .load_full()
+            .expect("Output must exist")
+            .output_status
+        {
+            ExecutionStatus::Success(t) | ExecutionStatus::SkipRest(t) => {
+                t.record_groups(finalized_groups);
+            },
+            ExecutionStatus::Abort(_) => {},
+        };
+    }
+
     // Called when a transaction is committed to record WriteOps for materialized aggregator values
-    // corresponding to the (deltas) in the recorded final output of the transaction.
-    pub(crate) fn record_delta_writes(
+    // corresponding to the (deltas) in the recorded final output of the transaction, as well as
+    // finalized (resource) group updates (which must be stored internally).
+    pub(crate) fn record_additional_writes(
         &self,
         txn_idx: TxnIndex,
         delta_writes: Vec<(T::Key, WriteOp)>,
```

Comment:
As we renamed the function from delta_writes to additional_writes, may be we should rename the input argument from delta_writes to additional_writes?

Reply:
these are still delta writes, additional because it now also incorporates group writes that must be previously recorded. Record here was confusing so it's now called incorporate_additional_writes.